org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/TypeReference.xtend

```diff
@@ -156,13 +156,17 @@ class TypeReference {
 				new QualifiedClassName('org.eclipse.emf.ecore', clazz.name)
 			}
 		} else {
-			new QualifiedClassName(GenModelUtil2.getGenClass(clazz, resourceSet).genPackage.qualifiedPackageName,
+			val genClass = GenModelUtil2.getGenClass(clazz, resourceSet)
+			val packageName = genClass.genPackage.getInterfacePackageName();
+			new QualifiedClassName(packageName,
 				GenModelUtil2.getGenClass(clazz, resourceSet).interfaceName)
```

Comment:
You could reuse the previously found genClass here instead of calling getGenClass again.

Reply:
done